src/ImageSharp/Formats/Png/PngDecoderCore.cs

```diff
@@ -1364,6 +1371,21 @@ private bool TryReadTextKeyword(ReadOnlySpan<byte> keywordBytes, out string name
             return true;
         }
 
+        private bool IsXmpTextData(ReadOnlySpan<byte> keywordBytes)
+        {
+            ReadOnlySpan<byte> expected = PngConstants.XmpKeyword;
+            bool result = keywordBytes.Length == expected.Length;
+            if (result)
+            {
+                for (int i = 0; i < keywordBytes.Length; i++)
+                {
+                    result |= keywordBytes[i] == expected[i];
+                }
+            }
+
+            return result;
```

Comment:
```suggestion
            return keywordBytes.SequenceEqual(expected);
```

Reply:
Good suggestion, made it into a one-liner now.